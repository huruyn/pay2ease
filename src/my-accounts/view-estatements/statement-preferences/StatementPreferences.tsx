import { Row, Table } from 'react-bootstrap';
import CommonPageContainer from '../../../common/CommonPageContainer';

export interface AccountType {
  id: number,
  account: string,
  number: string,
  alertStatus: string,
  statemetOption: string,
  contactMethods: string,
}

export default function StatementPreferences() {
  const Accounts: AccountType[] = [
    {
      id: 1,
      account: 'Chequing ',
      number: '01702-81-99639',
      statemetOption: 'eStatement',
      alertStatus: 'Inactive',
      contactMethods: 'Message Center Primary email (email@gmail.com)',
    },
    {
      id: 2,
      account: 'Savings ',
      number: '01702-81-99639',
      statemetOption: 'eStatement',
      alertStatus: 'Inactive',
      contactMethods: '',
    },
    {
      id: 3,
      account: 'Other ',
      number: '01702-81-99639',
      statemetOption: 'eStatement',
      alertStatus: 'Inactive',
      contactMethods: '',
    },
  ];
  return (
    <div className="page-content">
      <CommonPageContainer title="Statement Preferences">
        <Row style={{ justifyContent: 'space-between', marginTop: '15px' }}>
          <div style={{ marginTop: '20px' }}>
            <span>
              <h6>BANK ACCOUNTS</h6>
              <i />
            </span>
          </div>
          <hr style={
            {
              height: '3px',
              color: 'C4C4C4',
              margin: '0px',
            }
          }
          />
          <Row>
            <Table className="table table-borderless">
              <thead style={
              {
                height: '40px',
                background: '#EEEEEE',
                margin: '0px',
              }
            }
              >
                <tr>
                  <th scope="col" style={{ width: '25%' }}>Account</th>
                  <th scope="col" style={{ width: '25%' }}>STATEMENT OPTION</th>
                  <th scope="col" style={{ width: '15%' }}>{' '}</th>
                  <th scope="col" style={{ width: '35%' }}>
                    <span>
                      CONTACT METHODS FOR ALERTS
                      <i />
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody>
                {
                Accounts.map((account, index) => (
                  <tr key={account.id} style={{ borderBottom: index === Accounts.length - 1 ? '' : '1px solid #EEEEEE' }}>
                    <td>
                      <span style={{ marginTop: '20px' }}>
                        <h6>{account.account}</h6>
                        <h6>{`(${account.number})`}</h6>
                      </span>
                    </td>
                    <td>
                      <span>
                        <h6>{account.statemetOption}</h6>
                        <h6>{`(Alerts: ${account.alertStatus})`}</h6>
                      </span>
                    </td>
                    <td>
                      <h6><a href="./" style={{ color: 'black', verticalAlign: 'center' }}>Edit Preferences</a></h6>
                    </td>
                    <td>
                      <span style={{ marginTop: '20px' }}>
                        <h6>{account.contactMethods}</h6>
                      </span>
                    </td>

                  </tr>
                ))
              }
              </tbody>
            </Table>

          </Row>
          <hr style={{ height: '1px' }} />
          <button
            type="button"
            style={
            {
              width: '20%', alignSelf: 'flex-end', marginLeft: '80%', marginBottom: '10px', marginTop: '10px',
            }
          }
          >
            My Account
          </button>
          <hr style={{ height: '1px' }} />
          <Row>
            <i />
            <h6 style={{ fontWeight: 'bold' }}>Note:</h6>
            <ul>
              <li>
                An inactive alert status applies to bank accounts that were automatically
                transferred from Paperless to eStatements.
                If your alert status is displayed as inactive,
                it means you have not agreed to the terms and conditions for eStatements.
              </li>
              <li>
                If you opened your account within the last 30 days and requested mailed
                quarterly statements,you may still see Mailed Statement as your statement option.
                Your preferences will display correctly after the next statement cycle
              </li>
              <li>
                Changes to your preferences for bankbooks
                will not be reflected until processing is complete.
              </li>
              <li style={{ listStyle: 'none' }}>
                <i />
                Bank Accounts eStatement Terms and Conditions (May 2016) (PDF, 146 KB)
              </li>
              <li style={{ listStyle: 'none' }}>
                <i />
                Credit Card eStatement Terms and Conditions (May 2016) (PDF, 127 KB)
              </li>
              <li style={{ listStyle: 'none' }}>
                <i />
                Personal Line of Credit eStatement Terms and
                Conditions (PDF, 100 KB) Last updated October 2019
              </li>
              <li style={{ listStyle: 'none' }}>
                <i />
                ZippyX Securities Inc. and PPS Consent
                to Electronic Delivery (November 2014) (PDF, 47 KB)
              </li>
            </ul>
            <p>
              To view or save a eStatement as a PDF you will
              need Adobe Acrobat ReaderÂ°.
              Please download it before opening your eStatement.
            </p>
            <p>
              Portable Document Format (PDF) is a standard form of printable
              web document. With a PDF, you can save and print your
              statements for your personal record keeping.
            </p>
          </Row>
        </Row>
      </CommonPageContainer>
    </div>
  );
}
