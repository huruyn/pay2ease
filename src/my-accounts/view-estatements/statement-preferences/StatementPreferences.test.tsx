import React from 'react';
import StatementPreferences from './StatementPreferences';

const ReactTestRenderer = require('react-test-renderer');

describe('StatementPreferences Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <StatementPreferences />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
