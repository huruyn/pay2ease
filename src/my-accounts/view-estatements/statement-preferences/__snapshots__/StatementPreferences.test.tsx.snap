// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatementPreferences Component matches the snapshot 1`] = `
<div
  className="page-content"
>
  <div
    className="d-flex justify-content-between align-items-center"
  >
    <h3>
      Statement Preferences
    </h3>
  </div>
  <div
    className="content-wrapper"
  >
    <div
      className="row"
      style={
        Object {
          "justifyContent": "space-between",
          "marginTop": "15px",
        }
      }
    >
      <div
        style={
          Object {
            "marginTop": "20px",
          }
        }
      >
        <span>
          <h6>
            BANK ACCOUNTS
          </h6>
          <i />
        </span>
      </div>
      <hr
        style={
          Object {
            "color": "C4C4C4",
            "height": "3px",
            "margin": "0px",
          }
        }
      />
      <div
        className="row"
      >
        <table
          className="table table-borderless table"
        >
          <thead
            style={
              Object {
                "background": "#EEEEEE",
                "height": "40px",
                "margin": "0px",
              }
            }
          >
            <tr>
              <th
                scope="col"
                style={
                  Object {
                    "width": "25%",
                  }
                }
              >
                Account
              </th>
              <th
                scope="col"
                style={
                  Object {
                    "width": "25%",
                  }
                }
              >
                STATEMENT OPTION
              </th>
              <th
                scope="col"
                style={
                  Object {
                    "width": "15%",
                  }
                }
              >
                 
              </th>
              <th
                scope="col"
                style={
                  Object {
                    "width": "35%",
                  }
                }
              >
                <span>
                  CONTACT METHODS FOR ALERTS
                  <i />
                </span>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              style={
                Object {
                  "borderBottom": "1px solid #EEEEEE",
                }
              }
            >
              <td>
                <span
                  style={
                    Object {
                      "marginTop": "20px",
                    }
                  }
                >
                  <h6>
                    Chequing 
                  </h6>
                  <h6>
                    (01702-81-99639)
                  </h6>
                </span>
              </td>
              <td>
                <span>
                  <h6>
                    eStatement
                  </h6>
                  <h6>
                    (Alerts: Inactive)
                  </h6>
                </span>
              </td>
              <td>
                <h6>
                  <a
                    href="./"
                    style={
                      Object {
                        "color": "black",
                        "verticalAlign": "center",
                      }
                    }
                  >
                    Edit Preferences
                  </a>
                </h6>
              </td>
              <td>
                <span
                  style={
                    Object {
                      "marginTop": "20px",
                    }
                  }
                >
                  <h6>
                    Message Center Primary email (email@gmail.com)
                  </h6>
                </span>
              </td>
            </tr>
            <tr
              style={
                Object {
                  "borderBottom": "1px solid #EEEEEE",
                }
              }
            >
              <td>
                <span
                  style={
                    Object {
                      "marginTop": "20px",
                    }
                  }
                >
                  <h6>
                    Savings 
                  </h6>
                  <h6>
                    (01702-81-99639)
                  </h6>
                </span>
              </td>
              <td>
                <span>
                  <h6>
                    eStatement
                  </h6>
                  <h6>
                    (Alerts: Inactive)
                  </h6>
                </span>
              </td>
              <td>
                <h6>
                  <a
                    href="./"
                    style={
                      Object {
                        "color": "black",
                        "verticalAlign": "center",
                      }
                    }
                  >
                    Edit Preferences
                  </a>
                </h6>
              </td>
              <td>
                <span
                  style={
                    Object {
                      "marginTop": "20px",
                    }
                  }
                >
                  <h6>
                    
                  </h6>
                </span>
              </td>
            </tr>
            <tr
              style={
                Object {
                  "borderBottom": "",
                }
              }
            >
              <td>
                <span
                  style={
                    Object {
                      "marginTop": "20px",
                    }
                  }
                >
                  <h6>
                    Other 
                  </h6>
                  <h6>
                    (01702-81-99639)
                  </h6>
                </span>
              </td>
              <td>
                <span>
                  <h6>
                    eStatement
                  </h6>
                  <h6>
                    (Alerts: Inactive)
                  </h6>
                </span>
              </td>
              <td>
                <h6>
                  <a
                    href="./"
                    style={
                      Object {
                        "color": "black",
                        "verticalAlign": "center",
                      }
                    }
                  >
                    Edit Preferences
                  </a>
                </h6>
              </td>
              <td>
                <span
                  style={
                    Object {
                      "marginTop": "20px",
                    }
                  }
                >
                  <h6>
                    
                  </h6>
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <hr
        style={
          Object {
            "height": "1px",
          }
        }
      />
      <button
        style={
          Object {
            "alignSelf": "flex-end",
            "marginBottom": "10px",
            "marginLeft": "80%",
            "marginTop": "10px",
            "width": "20%",
          }
        }
        type="button"
      >
        My Account
      </button>
      <hr
        style={
          Object {
            "height": "1px",
          }
        }
      />
      <div
        className="row"
      >
        <i />
        <h6
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Note:
        </h6>
        <ul>
          <li>
            An inactive alert status applies to bank accounts that were automatically transferred from Paperless to eStatements. If your alert status is displayed as inactive, it means you have not agreed to the terms and conditions for eStatements.
          </li>
          <li>
            If you opened your account within the last 30 days and requested mailed quarterly statements,you may still see Mailed Statement as your statement option. Your preferences will display correctly after the next statement cycle
          </li>
          <li>
            Changes to your preferences for bankbooks will not be reflected until processing is complete.
          </li>
          <li
            style={
              Object {
                "listStyle": "none",
              }
            }
          >
            <i />
            Bank Accounts eStatement Terms and Conditions (May 2016) (PDF, 146 KB)
          </li>
          <li
            style={
              Object {
                "listStyle": "none",
              }
            }
          >
            <i />
            Credit Card eStatement Terms and Conditions (May 2016) (PDF, 127 KB)
          </li>
          <li
            style={
              Object {
                "listStyle": "none",
              }
            }
          >
            <i />
            Personal Line of Credit eStatement Terms and Conditions (PDF, 100 KB) Last updated October 2019
          </li>
          <li
            style={
              Object {
                "listStyle": "none",
              }
            }
          >
            <i />
            ZippyX Securities Inc. and PPS Consent to Electronic Delivery (November 2014) (PDF, 47 KB)
          </li>
        </ul>
        <p>
          To view or save a eStatement as a PDF you will need Adobe Acrobat Reader°. Please download it before opening your eStatement.
        </p>
        <p>
          Portable Document Format (PDF) is a standard form of printable web document. With a PDF, you can save and print your statements for your personal record keeping.
        </p>
      </div>
    </div>
  </div>
  <div
    className="footer-content-wrapper"
  >
    <div
      className="quick-links container"
    >
      <div
        className="row"
      >
        <h3
          className="col-sm-12"
        >
          Quick Links
        </h3>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-2"
        >
          <ul>
            <li>
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Home
              </a>
            </li>
            <li>
              Please Review Our Legal Stuff by visiting our
               
              <a
                className="text-decoration-none"
                href="/legal"
                onClick={[Function]}
              >
                Privacy Policy | TOU | EULA | CardHolder T&C Pages
              </a>
            </li>
          </ul>
        </div>
        <div
          className="col-sm-2"
        >
          8465 W Sahara Ave. Las Vegas, NEVADA
          <ul>
            <li>
              <a
                href="tel:+17026231876"
              >
                1.702.623.1876
              </a>
            </li>
            <li>
              <a
                href="mailto:contact@zippy.cash"
              >
                contact@zippy.cash
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12 col-sm-12 col-xs-12 text-left"
        >
          © 2021 Zippy Cash LLC  All rights reserved.
          <br />
          <br />
          The Program Manager for the Zippy Cash® Prepaid Visa Debit Card is PayQuicker, LLC
          <br />
          Zippy Cash® is a registered trademark of Zippy Cash, LLC
          <br />
          The Zippy Cash Prepaid Visa Debit Card is issued by Metropolitan Commercial Bank (Member FDIC) pursuant to a license from Visa U.S.A. Inc. “Metropolitan Commercial Bank” and “Metropolitan” are registered trademarks of Metropolitan Commercial Bank ©2014. BY USING THIS CARD YOU AGREE TO THE TERMS AND CONDITIONS OF THE CARDHOLDER AGREEMENT AND FEE SCHEDULE, IF ANY. All trademarks and service marks are the property of their respective owners.
          <br />
          In compliance with the USA PATRIOT Act* and other laws, anyone who is a US resident over 18 with a valid Social Security number and a verifiable physical US address (no PO boxes or APO addresses) can apply for an Zippy Cash Card. This information must be provided, along with other information pertinent to receiving approval, when applying for a Zippy Cash Card. Currently, we do not issue the Zippy Cash Card to residents of Puerto Rico, Guam, the Virgin Islands, American Samoa, or other US territories.
          <br />
          *Important information for opening a card account: To help the federal government fight the funding of terrorism and money laundering activities, the USA PATRIOT Act requires all financial institutions and their third parties to obtain, verify and record information that identifies each person who opens a card account. What this means for you: When you open a card account, we will ask for your name, address, date of birth, social security number or country identification number, and other information that will allows us to identify you. We may also request a copy of your driver’s license or other documentation bearing your photo as verification of your identity.
          <br />
          Apple and the Apple logo are trademarks of Apple Inc., registered in the U.S. and other countries. App Store is a service mark of Apple Inc. Google Play is a trademark of Google Inc. Data rates may apply. Samsung Pay is a trademark of Samsung Inc.
        </div>
      </div>
    </div>
    <footer
      className="container mb-3"
    >
      <div
        className="d-none d-md-block"
      >
        <div
          className="d-flex justify-content-between border-bottom"
        />
        <div
          className="d-flex justify-content-between border-top"
        >
          <div
            className=""
          >
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/personal-profile"
              >
                dev docs
              </a>
            </div>
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Lore
              </a>
            </div>
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Ipsum
              </a>
            </div>
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Nullam Atoidio
              </a>
            </div>
          </div>
          <div
            className="copyright mt-1"
          >
            <span>
              © 2021 Lorem ipsum dolor sit amet, consectetuer
            </span>
          </div>
        </div>
      </div>
      <div
        className="d-block d-md-none"
      >
        <div
          className="d-flex border-bottom"
        >
          <div
            className="d-flex flex-column me-2"
          >
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/personal-profile"
              >
                dev docs
              </a>
            </div>
            <div
              className="footer-link mb-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Lore
              </a>
            </div>
            <div
              className="footer-link mb-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Ipsum
              </a>
            </div>
            <div
              className="footer-link mb-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Nullam Atoidio
              </a>
            </div>
          </div>
        </div>
        <div
          className="d-flex justify-content-end border-top"
        >
          <div
            className="copyright mt-2 text-end small"
          >
            <span>
              © 2021 Lorem ipsum dolor sit amet, consectetuer
            </span>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
